### ISSUE TITLE
Chrome Perf Issue / Long Running Tasks

### ISSUE THREAD

**Issue by cdowdy** [feimosi/baguetteBox.js#84]
**device**: Nexus 5  

**browser** : Chrome Beta 50.0.2661.49  

**Description**:  
On an Android device running Chrome when you open any image lightbox you get met - randomly - with this in the console:  

```
Deferred long-running timer task(s) to improve scrolling smoothness. See crbug.com/574343  
```

here is the reference bug:  https://bugs.chromium.org/p/chromium/issues/detail?id=574343  

**Example URL**:  https://feimosi.github.io/baguetteBox.js/ 

**Example URL Data**:  

here is the timeline data from chrome -sorry for the dropbox link the file is rather large :(   
- https://dl.dropboxusercontent.com/u/19395935/TimelineRawData-baguetteBox-20160330T112520.json  

**Possible Fixes:**  
This can be fixed by replacing `setTimeout` in the script with `requestAnimationFrame`.  

rough example:  

``` javascript
requestAnimationFrame(function() {  
  overlay.className = 'visible';  
  if(options.afterShow)  
    options.afterShow();
});  
```

This though might cause other issues.  
1. [Browser Support](http://caniuse.com/#feat=requestanimationframe) is IE10 + and Android Browser 4.4+  
   - then again 2.0 is looking to drop ie < 11 https://github.com/feimosi/baguetteBox.js/issues/57
2. If you do go the RAF route a polyfill could make the script larger than you wish.  

**Possible Fix Results:**  
here is the timeline data from replacing the following setTimeouts with requestAnimationFrame  
1. https://github.com/feimosi/baguetteBox.js/blob/master/src/baguetteBox.js#L322  
2. https://github.com/feimosi/baguetteBox.js/blob/master/src/baguetteBox.js#L358  
3. https://github.com/feimosi/baguetteBox.js/blob/master/src/baguetteBox.js#L453  
4. https://github.com/feimosi/baguetteBox.js/blob/master/src/baguetteBox.js#L474  
5. https://github.com/feimosi/baguetteBox.js/blob/master/src/baguetteBox.js#L488  

and the timeline data ... again sorry about the DropBox link its a big file haha.  
- https://dl.dropboxusercontent.com/u/19395935/TimelineRawData-baguetteRAF-20160330T114254.json  

you can load the data in Chrome Devtools under Timeline > right click load Timeline data. Then you can filter the Event Data with "timer Fired" to get all the set timeout calls exceeding 50 milliseconds. Newer and/or more powerfull phones may not run into this and things may change in Chrome. This also may not be the appropriate "fix". I just wanted to make you aware!

**Comment by feimosi** [comment]
Thanks for pointing that out! Actually, I had no idea about it. 
`requestAnimationFrame` sounds like a good solution. For sure it's suitable for the milestone 2.0

**Comment by feimosi** [comment]
Closed in favor of #167.
